code/modules/food/recipes_microwave.dm

```diff
@@ -2151,6 +2156,11 @@
 	items = list(/obj/item/weapon/reagent_containers/food/snacks/chocolatebar)
 	result = /obj/item/weapon/reagent_containers/food/snacks/chococoin
 
+/datum/recipe/offcolorcheese
+	reagents = list(COLORFUL_REAGENT = 10) //wanted to use paint but can't find paint as a reagent proper and am not sure if the microwave can handle it
```

Comment:
Paint is a reagent, but it is not grouped with any other reagents because whoknows. You can find relevant code in `objects/items/weapons/paint.dm`. For your purposes though you only want the ID, which is `paint_blue`. This is not a define.

Reply:
Do I need to make paint_blue a proper define?
To be honest, I actually wanted to make the cheese a randomcolour dependent on paint colour (so you could use any paint), that colour overlay that honeycombs have, for example. But that was WAY outside my capacities, both coding and spriting-wise.

Reply:
No, you don't need a define in this special case.

Having it generate a color wouldn't be too hard if you do some magic in cooking code.

Reply:
How would that magic happen though. IF I somehow managed to do the magic, I'd also need to use said magic (and make the colour overlays) for all the cheese using recipes.

Reply:
For example, here's some offcolour lasagna and offcolour queso dip, recoloured for offcolour (blue) cheese. How would I go to make these use a random-colour instead of a designed colour? I'm guessing something like making recolour overlays for all the cheese using food, assigning a hue offset value if needed then using the foodcolour var that bread, cookies, soup, etc use.
![offcolour queso dip](https://user-images.githubusercontent.com/67024428/125367141-9eb76600-e345-11eb-8985-1363b5609966.png)
![offcolour lasagna](https://user-images.githubusercontent.com/67024428/125366987-4a13eb00-e345-11eb-8285-141cfd190579.png)

Reply:
I can provide you some advice on this, but yeah, you'd want to do a colored overlay. The way you would do this is first make a version of each of the cheesy items that you want and fully remove the yellow/blue part. Then make an overlay which is that exact part you removed (just paste it into a new spritesheet) and make it a grayscale version (basically, just different intensities of gray). Then you can color that overlay and stick it on the sprites. This is how reagents work.

You can stuff all of this code into the recipe's `make_food` child proc and then just call the parent (see recipe.dm L123 for details)